### PR TITLE
feat: added radio controls

### PR DESCRIPTION
**Pull request by alvarosabu**
### Description

As part of https://github.com/histoire-dev/histoire/issues/30 this PR adds a built-in select control <HstRadio />,

I added the possibility of passing the options as an array of objects { label: string, value: string} or as an array of values. Maybe an enhancement could be to pass a prop to define the key and the value, but I didn't want to complicate it. Both examples are available in the local story HstRadio.story.vue

I also added the control to the vue3 examples app

### Additional context

---

### What is the purpose of this pull request? 

- [ ] Bug fix
- [x] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] If it's a new feature, provide a convincing reason to add it. *Ideally, you should open a suggestion issue first and have it approved before working on it.*
- [x] Read the [Contributing Guidelines](https://github.com/Akryum/histoire/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/Akryum/histoire/blob/main/CONTRIBUTING.md#pull-request-guidelines) and follow the [Commit Convention](https://github.com/Akryum/histoire/blob/main/.github/commit-convention.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

**Review by Akryum — CHANGES_REQUESTED** [review]
image

Looking great! A few issues:
- Alignment is a bit off
- Border color doesn't match the other input borders

File: packages/histoire-controls/src/components/radio/HstRadio.vue [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+export interface RadioOption {
+  label: string
+  value: string
+}
+
+const props = defineProps<{
+  title?: string
+  modelValue: string
+  options: RadioOption[]
```
Comment (hugoattal) [inline packages/histoire-controls/src/components/radio/HstRadio.vue:19]: Wouldn't it be more consistent to have the same type as the Select component 🤔 ?
https://github.com/histoire-dev/histoire/blob/2a64cb52d816b337415211fb29f2c3cc1943776e/packages/histoire-controls/src/components/select/HstSelect.vue#L17

File: packages/histoire-controls/src/components/radio/HstRadio.vue [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+          height="16"
+          viewBox="0 0 24 24"
+          class="htw-relative htw-z-10 htw-border htw-border-solid group-active:htw-bg-gray-500/20 htw-text-inherit htw-rounded-full htw-box-border htw-inset-0 htw-transition-border htw-duration-150 htw-ease-out htw-mr-2"
+        >
+          <circle
+            cx="12"
+            cy="12"
+            r="8"
+            :class="[
+              animationEnabled ? 'htw-transition-all' : 'htw-transition-none',
```
Comment (hugoattal) [inline packages/histoire-controls/src/components/radio/HstRadio.vue:89]: Why not always enabling the animation? I'm not sure to understand the necessity of the `animationEnabled` boolean 😅...

**Review by hugoattal — COMMENTED** [review]
Thank you so much for your contribution ❤️!

Nice work, I just have a few questions 🙂

File: packages/histoire-controls/src/components/radio/HstRadio.vue (hunk 1 above)
Comment (alvarosabu) [reply packages/histoire-controls/src/components/radio/HstRadio.vue:19]: Totally agree with you, I was hesitating if adding it or not, where could we add types and interfaces that are global? a `src/types.ts` file?

File: packages/histoire-controls/src/components/radio/HstRadio.vue (hunk 2 above)
Comment (alvarosabu) [reply packages/histoire-controls/src/components/radio/HstRadio.vue:89]: I kinda copy and paste that part from the checkbox, but not sure if we need it on the radios

File: packages/histoire-controls/src/components/radio/HstRadio.vue (hunk 1 above)
Comment (hugoattal) [reply packages/histoire-controls/src/components/radio/HstRadio.vue:19]: `src/types.ts` looks fine to me 👍 !

File: packages/histoire-controls/src/components/radio/HstRadio.vue (hunk 2 above)
Comment (hugoattal) [reply packages/histoire-controls/src/components/radio/HstRadio.vue:89]: Oh! Wdyt @Akryum?

Comment (Akryum) [reply packages/histoire-controls/src/components/radio/HstRadio.vue:89]: It's to prevent the animations from playing on mounted. Feel free to add comments here and in the checkbox component 👍

File: packages/histoire-controls/src/components/radio/HstRadio.vue (hunk 1 above)
Comment (alvarosabu) [reply packages/histoire-controls/src/components/radio/HstRadio.vue]: Done :D

File: packages/histoire-controls/src/components/radio/HstRadio.vue (hunk 2 above)
Comment (alvarosabu) [reply packages/histoire-controls/src/components/radio/HstRadio.vue:90]: Done

File: packages/histoire-controls/src/components/checkbox/HstCheckbox.vue [hunk 3]
```diff
@@ -14,7 +14,7 @@ const props = defineProps<{
 }>()
 
 const emit = defineEmits({
-  'update:modelValue': (newValue: boolean) => true,
+  'update:modelValue': (_newValue: boolean) => true,
```
Comment (Akryum) [inline packages/histoire-controls/src/components/checkbox/HstCheckbox.vue:17]: Why this change?

Comment (alvarosabu) [reply packages/histoire-controls/src/components/checkbox/HstCheckbox.vue:17]: Linter was complaining, newValue is not being used, is just returning true.

We probably could get rid of the param but not sure why was there tho.

Comment (Akryum) [reply packages/histoire-controls/src/components/checkbox/HstCheckbox.vue:17]: It's here for type inference, but maybe you can turn it to the Generic version of `defineEmits`

**Review by Akryum — APPROVED** [review]
